pkg/lib/authenticationflow/errors.go

```diff
@@ -29,6 +29,12 @@ var ErrNoChange = errors.New("no change")
 // This error originates from CanReactTo and will be propagated to public API.
 var ErrEOF = errors.New("eof")
 
+// ErrBotProtectionVerificationSomeStatus means bot-protection verification is of some status
+// This error can only be returned by IntentBotProtection ReactTo
+var ErrBotProtectionVerificationFailed = errors.New("bot-protection verification failed")
+var ErrBotProtectionVerificationSuccess = errors.New("bot-protection verification success")
+var ErrBotProtectionVerificationServiceUnavailable = errors.New("bot-protection verification service unavailable")
+
```

Comment:
Never use `errors.New` to construct errors that we will report in the API. `errors.New` appears as unexpected errors in the API. We must use apierror to report expected errors.

Reply:
> that we will report in the API

These errors are not actual errors. They are not expected to be visible to users.

They acts similarly to `ErrEOF` to allow special handling in in caller.

Caller can then mutate the context/session on verification status. 

Note - the actual errors are in [pkg/lib/feature/botprotection/errors.go](https://github.com/authgear/authgear-server/pull/4390/files/8010123425bd85aad19dd9937e09514536e92698..911969044381271e059d16afdb749b5d302fb90b#diff-b17e6ea1c38095c33b5d8f42eba53871f909b88f90f74de31994c624a6ac479a)

Any thoughts?